src/task_system/task_system_server/Controllers/TaskSystemController.cs

```diff
@@ -44,24 +44,53 @@ public async Task<IActionResult> GetTasks([FromQuery] QueryObject query)
 
         //GET: /tasks/{id}
         [HttpGet("{id}")]
-        public async Task<IActionResult> GetTaskById([FromRoute] int id)
+        public async Task<IActionResult> GetTaskById([FromRoute] int taskId)
         {
             if (!Request.Headers.TryGetValue("X-API-Key", out var apiKey) || !_permissionValidator.ValidatePermissions(PermissionNames.VIEW_TASKS, apiKey!))
                 return Unauthorized(new TaskSystemResponse<string>(ResponseMessages.UNAUTHORIZED, null));
 
             if (!ModelState.IsValid)
                 return BadRequest(ModelState);
 
-            var task = await _taskSystemRepository.GetTaskByIdAsync(id);
+            var task = await _taskSystemRepository.GetTaskByIdAsync(taskId);
 
             if (task == null)
             {
-                return NotFound(new TaskSystemResponse<int>(ResponseMessages.GET_TASK_FAILED, id));
+                return NotFound(new TaskSystemResponse<int>(ResponseMessages.GET_TASK_FAILED, taskId));
             }
 
             return Ok(new TaskSystemResponse<TaskItem>(ResponseMessages.GET_TASK_SUCCESS, task));
         }
 
+        //GET: /tasks/account/{accountId}
+        [HttpGet("account/{accountId}")]
```

Comment:
## Non-blocking
I'm not sure this endpoint naming is ideal.

- Main concern: There are 2 ways to get tasks depending on the permissions available to the user. It would be better if we get all tasks through a single endpoint.

- Pedantic concern: HTTP nouns should indicate the resource or sub-resource being accessed.
In this case, "task/account" implies that we are explicitly and only getting tasks' accounts data.

### Solution
A more standard way to handle this case would be to modify the existing "GET /tasks" endpoint to do the following:
1) Collect all tasks that the current account has permission to access. (in this case, that would be all tasks if they have "ViewTasks" permission, else only the tasks requested by their account)
2) Filter tasks using query string filtering (URL parameters) if needed (minimally, implement the "accountId={id}" filter)

### Note
However, you've already done the work, and it functions, so it's okay if you don't want to modify it since the project is nearly done.